File: CS/DragAppointmentExample/Form1.cs
using DevExpress.XtraScheduler;
using System;
using System.Windows.Forms;

namespace CodeExampleTemplate {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            this.schedulerControl1.Start = DateTime.Now.Date;
            this.schedulerControl1.DayView.TopRowTime = new TimeSpan(9, 0, 0);
            #region #events
            // Subscribe to events
            this.schedulerControl1.AppointmentDrag += SchedulerControl_AppointmentDrag;
            this.schedulerControl1.AdditionalAppointmentsDrag += SchedulerControl1_AdditionalAppointmentsDrag;
            #endregion #events
        }

        private void Form1_Load(object sender, EventArgs e) {
            InitHelper helper = new InitHelper(this.schedulerStorage1);

            schedulerStorage1.BeginUpdate();
            try {
                schedulerStorage1.Resources.DataSource = helper.InitResources();
                schedulerStorage1.Appointments.DataSource = helper.InitAppointments();
            }
            finally {
                schedulerStorage1.EndUpdate();
            }
            schedulerControl1.GroupType = DevExpress.XtraScheduler.SchedulerGroupType.Resource;
        }
        #region #eventhandlers
        private void SchedulerControl_AppointmentDrag(object sender, AppointmentDragEventArgs e) {
            // Find a special appointment among visible appointments.
            AppointmentBaseCollection visbileApts = schedulerControl1.ActiveView.GetAppointments();
            Appointment apt = visbileApts.Find(item => (item.Subject == "Finalize" && item.ResourceId.Equals(e.HitResource.Id)));
            if (apt == null) {
                e.AdditionalAppointments.Clear();
            }
            else {
                if (e.SourceAppointment != apt) {
                    // If the dragged appointment overlaps with the special appointment,
                    // add the special appointment to the AdditionalAppointments collection
                    // to drag along with the initial appointment.
                    // The AdditionalAppointmentsDrag event will occur.
                    if (e.EditedAppointment.End > apt.Start)
                        e.AdditionalAppointments.Add(apt);
                    else
                        e.AdditionalAppointments.Clear();
                }
            }
        }

        private void SchedulerControl1_AdditionalAppointmentsDrag(object sender, AdditionalAppointmentsDragEventArgs e) {
            if (e.AdditionalAppointmentInfos.Count <= 0)
                return;
            AppointmentDragInfo aptInfo = e.AdditionalAppointmentInfos[0];
            // If the end of the main dragged appointment becomes greater than the start of the special appointment,
            // set the start of the special appointment being dragged to the end of the main dragged appointment.
            // It prevents overlapping appointments.
            if (e.PrimaryAppointmentInfos[0].EditedAppointment.End > aptInfo.SourceAppointment.Start)
                aptInfo.EditedAppointment.Start = e.PrimaryAppointmentInfos[0].EditedAppointment.End;
        }
        #endregion #eventhandlers
    }
}

File: VB/DragAppointmentExample/Form1.vb
Imports DevExpress.XtraScheduler
Imports System
Imports System.Windows.Forms

Namespace CodeExampleTemplate
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            Me.schedulerControl1.Start = Date.Now.Date
            Me.schedulerControl1.DayView.TopRowTime = New TimeSpan(9, 0, 0)
'            #Region "#events"
            ' Subscribe to events
            AddHandler Me.schedulerControl1.AppointmentDrag, AddressOf SchedulerControl_AppointmentDrag
            AddHandler Me.schedulerControl1.AdditionalAppointmentsDrag, AddressOf SchedulerControl1_AdditionalAppointmentsDrag
'            #End Region ' #events
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim helper As New InitHelper(Me.schedulerStorage1)

            schedulerStorage1.BeginUpdate()
            Try
                schedulerStorage1.Resources.DataSource = helper.InitResources()
                schedulerStorage1.Appointments.DataSource = helper.InitAppointments()
            Finally
                schedulerStorage1.EndUpdate()
            End Try
            schedulerControl1.GroupType = DevExpress.XtraScheduler.SchedulerGroupType.Resource
        End Sub
        #Region "#eventhandlers"
        Private Sub SchedulerControl_AppointmentDrag(ByVal sender As Object, ByVal e As AppointmentDragEventArgs)
            ' Find a special appointment among visible appointments.
            Dim visbileApts As AppointmentBaseCollection = schedulerControl1.ActiveView.GetAppointments()
            Dim apt As Appointment = visbileApts.Find(Function(item) (item.Subject = "Finalize" AndAlso item.ResourceId.Equals(e.HitResource.Id)))
            If apt Is Nothing Then
                e.AdditionalAppointments.Clear()
            Else
                If e.SourceAppointment IsNot apt Then
                    ' If the dragged appointment overlaps with the special appointment,
                    ' add the special appointment to the AdditionalAppointments collection
                    ' to drag along with the initial appointment.
                    ' The AdditionalAppointmentsDrag event will occur.
                    If e.EditedAppointment.End > apt.Start Then
                        e.AdditionalAppointments.Add(apt)
                    Else
                        e.AdditionalAppointments.Clear()
                    End If
                End If
            End If
        End Sub

        Private Sub SchedulerControl1_AdditionalAppointmentsDrag(ByVal sender As Object, ByVal e As AdditionalAppointmentsDragEventArgs)
            If e.AdditionalAppointmentInfos.Count <= 0 Then
                Return
            End If
            Dim aptInfo As AppointmentDragInfo = e.AdditionalAppointmentInfos(0)
            ' If the end of the main dragged appointment becomes greater than the start of the special appointment,
            ' set the start of the special appointment being dragged to the end of the main dragged appointment.
            ' It prevents overlapping appointments.
            If e.PrimaryAppointmentInfos(0).EditedAppointment.End > aptInfo.SourceAppointment.Start Then
                aptInfo.EditedAppointment.Start = e.PrimaryAppointmentInfos(0).EditedAppointment.End
            End If
        End Sub
        #End Region ' #eventhandlers
    End Class
End Namespace
